src/uct/tcp/tcp_ep.c

```diff
@@ -844,6 +845,12 @@ static inline void uct_tcp_ep_handle_put_req(uct_tcp_ep_t *ep,
     ep->rx.offset += copied_length;
     ep->rx.put_sn  = put_req->sn;
 
+    /* Remove the flag that indicates that EP is sending PUT RX ACK in order
+     * to not ack the uncompleted PUT RX operation for which PUT REQ is being
+     * handled here. ACK for both operations will be sent after the completion
+     * of the last received PUT operation */
+    ep->ctx_caps &= ~UCS_BIT(UCT_TCP_EP_CTX_TYPE_PUT_RX_SENDING_ACK);
```

Comment:
move it to uct_tcp_ep_handle_put_req(), after `ep->rx.put_sn  = put_req->sn;` :
we should prevent sending ACK after we increment the ep->tx.put_sn, until all data is received.

Reply:
@yosefe this is already there

Reply:
right, sorry

Reply:
> right, sorry

;)